### PR TITLE
feat(lib): Harmonize and simplify color objects

### PR DESCRIPTION
**Pull request by jacques-lebourgeois**
### Related issues

[Library: Harmonize our color objects](https://github.com/Orange-OpenSource/ods-charts/issues/291)

### Description

Simplify colors management. 

There was two set of colors `ODSChartsCategoricalColorsSet` and `ODSChartsSequentialColorsSet`.

There is now one set of Color only `ODSChartsColorsSet` that is used bith for Charts Colors and Visual Map Colors.

### Motivation & Context

It is compliest and remove the previous necessity to maintain colors set twice.

### Types of change

- Breaking change (fix or feature that would change existing functionality)
  The previous colors sets `ODSChartsCategoricalColorsSet` and `ODSChartsSequentialColorsSet` didn't exist anymore. Their utilisation must be replaced by `ODSChartsColorsSet`.

### Test checklist

Please check that the following tests projects are still working:

- [X] `docs/examples`
- [X] `test/angular-ngx-echarts`
- [X] `test/angular-echarts`
- [X] `test/html`
- [X] `test/react`
- [x] `test/vue`
- [x] `test/examples/bar-line-chart`
- [x] `test/examples/single-line-chart`
- [x] `test/examples/timeseries-chart`

### CODE REVIEW

File: src/theme/css-themes/css-themes.ts [hunk 1]
```diff
@@ -92,7 +92,9 @@ export class ODSChartsCSSPopoverDefinition extends ODSChartsItemGroupCSSDefiniti
 }
 
 /**
- * ODSChartsCSSThemeDefinition defines the classes an styles to be add on code generated by the library.
+ * `ODSChartsCSSThemeDefinition` defines the classes an styles to be add on code generated by the library.
```
Comment (louismaximepiton) [inline src/theme/css-themes/css-themes.ts:95]:
```suggestion
 * `ODSChartsCSSThemeDefinition` defines the classes and styles to be added on code generated by the library.
```

File: src/theme/ods-chart-theme.ts [hunk 2]
```diff
@@ -206,37 +188,43 @@ const THEMES: {
...
+      yellow: LIGHT_COLORS_YELLOW,
     },
-    sequentialColors: {
-      blue: LIGHT_SEQUENTIAL_COLORS_BLUE,
-      green: LIGHT_SEQUENTIAL_COLORS_GREEN,
-      pink: LIGHT_SEQUENTIAL_COLORS_PINK,
-      purple: LIGHT_SEQUENTIAL_COLORS_PURPLE,
-      yellow: LIGHT_SEQUENTIAL_COLORS_YELLOW,
+    visualMapColors: {
+      default: { visualMapColor: LIGHT_COLORS.color },
```
Comment (louismaximepiton) [inline src/theme/ods-chart-theme.ts:217]: I can't remember what we've said to do for the visual map colors, do we want all the colors ?

File: src/theme/ods-chart-theme.ts [hunk 3]
```diff
@@ -559,7 +551,7 @@ export class ODSChartsTheme {
    *
    *   {@link ODSChartsPopoverManagers} gives preconfigured renderer
    *
-   *   default value is  {@link ODSChartsPopoverManagers.NONE}: uses default Apache ECharts template to externalize tooltip/popover HTML element, implementing Orange Design system
+   *   The default value of is `ODSChartsPopoverManagers.NONE`: uses default Apache ECharts template to externalize tooltip/popover HTML element, implementing Orange Design system
```
Comment (louismaximepiton) [inline src/theme/ods-chart-theme.ts:554]:
```suggestion
   *   The default value of this parameter is `ODSChartsPopoverManagers.NONE`: uses default Apache ECharts template to externalize tooltip/popover HTML element, implementing Orange Design system
```

File: test/react/package-lock.json
Comment (louismaximepiton) [inline test/react/package-lock.json:1]: Why are there some changes in here ? I think it should either be in another PR or totally removed.

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
The PR lokos good! Here are some small comments.

File: src/theme/ods-chart-theme.ts (hunk 2 above)
Comment (jacques-lebourgeois) [reply src/theme/ods-chart-theme.ts:217]: I don't think we've decided anything. 

In the first version, I chose to include only color palettes of the same color. 

But now that we've simplified things by mutualizing color management, I don't see much reason to restrict it. Let's keep it simple, shall we?

File: test/react/package-lock.json
Comment (jacques-lebourgeois) [reply test/react/package-lock.json:1]: It's probably a side effect between my environments. I'll try to fix it.

Comment (jacques-lebourgeois) [reply test/react/package-lock.json:1]: [Revert package-lock changes](https://github.com/Orange-OpenSource/ods-charts/pull/412/commits/91023074b5ba0160a721aecb4365aa2177b43717)

File: src/theme/ods-chart-theme.ts (hunk 2 above)
Comment (louismaximepiton) [reply src/theme/ods-chart-theme.ts:217]: You're right :)

**Review by louismaximepiton — COMMENTED** [review]
Love it, thanks ! Let's :shipit:

**Review by louismaximepiton — APPROVED** [review]
